api/src/test/java/org/pmiops/workbench/api/UserControllerTest.java

```diff
@@ -48,66 +52,69 @@
 @AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
 public class UserControllerTest {
 
-  @Autowired private UserDao userDao;
-  @Mock private AdminActionHistoryDao adminActionHistoryDao;
-  @Mock private Provider<WorkbenchConfig> configProvider;
-  @Mock private Provider<User> userProvider;
-  @Mock private FireCloudService fireCloudService;
-  @Mock private ComplianceService complianceService;
-  @Mock private DirectoryService directoryService;
-  private UserService userService;
-
-  private UserController userController;
-  private Long incrementedUserId = 1L;
   private static final FakeClock CLOCK = new FakeClock(Instant.now(), ZoneId.systemDefault());
+  private static WorkbenchConfig config = new WorkbenchConfig();
+  private static User user = new User();
+  private static long incrementedUserId = 1;
+
+  @TestConfiguration
+  @Import({UserController.class, UserService.class})
+  @MockBean({
+    FireCloudService.class,
+    ComplianceService.class,
+    DirectoryService.class,
+    AdminActionHistoryDao.class
+  })
+  static class Configuration {
+
+    @Bean
+    @Scope("prototype")
+    public WorkbenchConfig workbenchConfig() {
+      return config;
+    }
+
+    @Bean
+    Clock clock() {
+      return CLOCK;
+    }
+
+    @Bean
+    @Scope("prototype")
+    User user() {
+      return user;
+    }
 
-  public Clock clock() {
-    return CLOCK;
+    @Bean
+    Random random() {
+      return new FakeLongRandom(123);
+    }
   }
 
+  @Autowired UserController userController;
+  @Autowired UserDao userDao;
+  @Autowired FireCloudService fireCloudService;
+
   @Before
   public void setUp() {
-    WorkbenchConfig config = new WorkbenchConfig();
     config.firecloud = new WorkbenchConfig.FireCloudConfig();
     config.firecloud.enforceRegistered = false;
     config.featureFlags = new FeatureFlagsConfig();
     config.featureFlags.useBillingProjectBuffer = false;
-    configProvider = Providers.of(config);
-    this.userService =
-        new UserService(
-            userProvider,
-            userDao,
-            adminActionHistoryDao,
-            clock(),
-            new Random(),
-            fireCloudService,
-            configProvider,
-            complianceService,
-            directoryService);
-    this.userController = new UserController(userService);
+
     saveFamily();
   }
 
-  @After
-  public void tearDown() {
-    userDao.deleteAll();
+  @Test(expected = ForbiddenException.class)
+  public void testEnforceRegisteredUnregistered() {
```

Comment:
Does it make sense to test when enforce register is false as well

Reply:
Yes - the existing tests cover this (by default, enforceRegistered=false in these tests).